qiskit_addon_sqd/fermion.py

```diff
@@ -28,6 +29,43 @@
 config.update("jax_enable_x64", True)  # To deal with large integers
 
 
+@dataclass(frozen=True)
+class SCIState:
+    """The amplitudes and determinants describing a quantum state.
+
+    ``amplitudes`` is an ``MxN`` array such that ``M`` = ``len(ci_strs_a)``
+    and ``N`` = ``len(ci_strs_b)``. ``amplitudes[i][j]`` corresponds to the
+    amplitude of the determinant pair (``ci_strs_a[i]``, ``ci_strs_b[j]``).
+    """
+
+    amplitudes: np.ndarray
+    ci_strs_a: np.ndarray
+    ci_strs_b: np.ndarray
+
+    def __post_init__(self):
+        """Validate dimensions of inputs."""
+        object.__setattr__(
+            self, "amplitudes", np.asarray(self.amplitudes)
+        )  # Convert to ndarray if not already
```

Comment:
Why are only the amplitudes converted? Can we just omit this, and leave it up to the user to pass an object of the correct type?

Reply:
the `numpy` api isn't used for this check on the other two structures.

This allows for the case that the user used some other Sequence besides np.arrays, I don't necessarily want this to break if users instantiate this with `list[list[float]]` `list[int]`, `list[int]`

Reply:
Ok